consensus/src/network_interface.rs

```diff
@@ -107,14 +142,19 @@ pub fn network_endpoint_config() -> AppConfig {
 }
 
 impl NewNetworkSender for ConsensusNetworkSender {
-    /// Returns a Sender that only sends for the `CONSENSUS_DIRECT_SEND_PROTOCOL` and
-    /// `CONSENSUS_RPC_PROTOCOL` ProtocolId.
     fn new(
+        node_config: Option<NodeConfig>,
```

Comment:
I don't think we should pass a node config here, it looks weird. it probably should expects a vec for direct send and another for rpc. the caller should be responsible for constructing that

Reply:
Interesting... I'm not sure I'd agree. In the ideal case, I don't want to pass this to the network sender at all 😄 I'd much rather find a way of not impacting the interface. Either way, I think we agree that this is not super nice and it's something we can clean up later. 😄 Thoughts?